src/components/ToggleSwitch/ToggleSwitch.spec.js

```diff
@@ -0,0 +1,28 @@
+import ToggleSwitch from './TogleSwitch';
+
+const { render, screen } = require('@testing-library/react');
+
+describe('ToggleSwitch', () => {
+  let container;
+
+  beforeEach(() => {
+    container = render(<ToggleSwitch />).container;
+  });
+  test('Should renders a ToggleStwitch component', () => {
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should contains a label with a switch role', () => {
+    const labelElement = screen.getByRole('switch');
+    expect(labelElement).toBeInTheDocument();
+  });
+  test('Should contains an input type checkbox', () => {
+    const inputElement = screen.getByTestId('input-checkbox');
```

Comment:
Remember that querying byTestId should be our last option, ideally we want to query byRole (as you did in other cases). You can see the recommended priority guide here: https://testing-library.com/docs/queries/about/#priority

Reply:
Thanks @moyomc ! I've added a button role  to my input.